src/Collapse.js

```diff
@@ -78,10 +84,36 @@ class Collapse extends Component {
     // else: do nothing.
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.collapse === SHOWN &&
+        prevState &&
+        prevState.collapse !== SHOWN) {
+      this.onOpened();
+    }
+
+    if (this.state.collapse === HIDDEN &&
+        prevState &&
+        prevState.collapse !== HIDDEN) {
+      this.onClosed();
+    }
+  }
+
   componentWillUnmount() {
     clearTimeout(this.transitionTag);
   }
 
+  onOpened() {
+    if (this.props.onOpened) {
+      this.props.onOpened();
```

Comment:
What do you think about defaulting these props to noop? (`() => {}`) then there would not be a need for these helper function nor the need to bind them in the constructor.